From: Justin Michaud <jmichaud@igalia.com>
Subject: Fix FTBFS in i386 and armhf
Bug: https://bugs.webkit.org/show_bug.cgi?id=299018
Origin: https://github.com/WebKit/WebKit/commit/fcaa289f6015595a638eb96b9f728eaaa7b13ab8
Index: webkitgtk/Source/JavaScriptCore/llint/LowLevelInterpreter32_64.asm
===================================================================

Upstream-Status: Backport [https://bugs.webkit.org/show_bug.cgi?id=299018]
--- webkitgtk.orig/Source/JavaScriptCore/llint/LowLevelInterpreter32_64.asm
+++ webkitgtk/Source/JavaScriptCore/llint/LowLevelInterpreter32_64.asm
@@ -3354,6 +3354,9 @@ op(loop_osr_entry_gate, macro ()
     crash() # Should never reach here.
 end)
 
+op(op_instanceof_return_location, macro ()
+    crash() # Should never reach here.
+end)
 
 llintOpWithMetadata(op_check_private_brand, OpCheckPrivateBrand, macro (size, get, dispatch, metadata, return)
     metadata(t5, t2)

