Index: wpewebkit-2.21.91/Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp
===================================================================
--- wpewebkit-2.21.91.orig/Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp
+++ wpewebkit-2.21.91/Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp
@@ -100,6 +100,7 @@ void WebFullScreenManager::videoControls
 
 void WebFullScreenManager::setPIPStandbyElement(WebCore::HTMLVideoElement* pipStandbyElement)
 {
+#if ENABLE(VIDEO)
     if (pipStandbyElement == m_pipStandbyElement)
         return;
 
@@ -110,6 +111,7 @@ void WebFullScreenManager::setPIPStandby
 
     if (m_pipStandbyElement)
         m_pipStandbyElement->setVideoFullscreenStandby(true);
+#endif
 }
 
 void WebFullScreenManager::didReceiveMessage(IPC::Connection& connection, IPC::Decoder& decoder)
