Index: wpewebkit-2.21.91/Source/WebCore/platform/GStreamer.cmake
===================================================================
--- wpewebkit-2.21.91.orig/Source/WebCore/platform/GStreamer.cmake
+++ wpewebkit-2.21.91/Source/WebCore/platform/GStreamer.cmake
@@ -36,7 +36,6 @@ if (ENABLE_VIDEO OR ENABLE_WEB_AUDIO)
         platform/mediastream/libwebrtc/GStreamerVideoDecoderFactory.cpp
         platform/mediastream/libwebrtc/GStreamerVideoEncoderFactory.cpp
         platform/mediastream/libwebrtc/LibWebRTCAudioModule.cpp
-        platform/mediastream/libwebrtc/LibWebRTCProviderGlib.cpp
 
         platform/mediastream/gstreamer/GStreamerAudioCaptureSource.cpp
         platform/mediastream/gstreamer/GStreamerAudioCapturer.cpp
Index: wpewebkit-2.21.91/Source/WebCore/platform/SourcesGLib.txt
===================================================================
--- wpewebkit-2.21.91.orig/Source/WebCore/platform/SourcesGLib.txt
+++ wpewebkit-2.21.91/Source/WebCore/platform/SourcesGLib.txt
@@ -35,3 +35,5 @@ platform/glib/SharedBufferGlib.cpp
 platform/glib/UserAgentGLib.cpp
 
 platform/network/glib/NetworkStateNotifierGLib.cpp
+
+platform/mediastream/libwebrtc/LibWebRTCProviderGlib.cpp
Index: wpewebkit-2.21.91/Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProviderGlib.cpp
===================================================================
--- wpewebkit-2.21.91.orig/Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProviderGlib.cpp
+++ wpewebkit-2.21.91/Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProviderGlib.cpp
@@ -41,7 +41,11 @@ UniqueRef<LibWebRTCProvider> LibWebRTCPr
 
 bool LibWebRTCProvider::webRTCAvailable()
 {
+#if USE(LIBWEBRTC) && USE(GSTREAMER)
     return true;
+#else
+    return false;
+#endif
 }
 
 #if USE(LIBWEBRTC) && USE(GSTREAMER)
