--- webkitgtk-2.4.11.orig/Source/WebKit2/Shared/API/c/WKString.h.orig   2017-12-17 20:37:47.918238974 +0100
+++ webkitgtk-2.4.11.orig/Source/WebKit2/Shared/API/c/WKString.h        2017-12-17 20:39:10.970532449 +0100
@@ -28,6 +28,7 @@
 
 #include <WebKit2/WKBase.h>
 #include <stddef.h>
+#include <uchar.h>
 
 #ifndef __cplusplus
 #include <stdbool.h>
@@ -39,7 +40,7 @@
 
 #if !defined(WIN32) && !defined(_WIN32) \
     && !((defined(__CC_ARM) || defined(__ARMCC__)) && !defined(__linux__)) /* RVCT */
-    typedef unsigned short WKChar;
+    typedef char16_t WKChar;
 #else
     typedef wchar_t WKChar;
 #endif
