--- wpewebkit-1.20170926-r0.orig/Source/JavaScriptCore/API/JSStringRef.h	2016-02-27 11:14:00.000000000 +0100
+++ wpewebkit-1.20170926-r0/Source/JavaScriptCore/API/JSStringRef.h	2017-02-27 11:14:05.000000000 +0100
@@ -27,6 +27,7 @@
 #define JSStringRef_h

 #include <JavaScriptCore/JSValueRef.h>
+#include <uchar.h>

 #ifndef __cplusplus
 #include <stdbool.h>
@@ -46,7 +47,7 @@
  character. As with all scalar types, endianness depends on the underlying
  architecture.
 */
-    typedef unsigned short JSChar;
+    typedef char16_t JSChar;
 #else
     typedef wchar_t JSChar;
 #endif
