diff -ruN git/Source/JavaScriptCore/API/JSStringRef.cpp git.patched/Source/JavaScriptCore/API/JSStringRef.cpp
--- git/Source/JavaScriptCore/API/JSStringRef.cpp       2018-02-27 16:47:25.182835822 +0100
+++ git.patched/Source/JavaScriptCore/API/JSStringRef.cpp       2018-02-27 16:58:07.649218536 +0100
@@ -37,7 +37,7 @@
 JSStringRef JSStringCreateWithCharacters(const JSChar* chars, size_t numChars)
 {
     initializeThreading();
-    return &OpaqueJSString::create(chars, numChars).leakRef();
+    return &OpaqueJSString::create(reinterpret_cast<const UChar*>(chars), numChars).leakRef();
 }

 JSStringRef JSStringCreateWithUTF8CString(const char* string)
@@ -62,7 +62,7 @@
 JSStringRef JSStringCreateWithCharactersNoCopy(const JSChar* chars, size_t numChars)
 {
     initializeThreading();
-    return OpaqueJSString::create(StringImpl::createWithoutCopying(chars, numChars)).leakRef();
+    return OpaqueJSString::create(StringImpl::createWithoutCopying(reinterpret_cast<const UChar*>(chars), numChars)).leakRef();
 }

 JSStringRef JSStringRetain(JSStringRef string)
@@ -87,7 +87,7 @@
 {
     if (!string)
         return nullptr;
-    return string->characters();
+    return reinterpret_cast<const JSChar*>(string->characters());
 }

 size_t JSStringGetMaximumUTF8CStringSize(JSStringRef string)
diff -ruN git/Source/JavaScriptCore/runtime/DateConversion.cpp git.patched/Source/JavaScriptCore/runtime/DateConversion.cpp
--- git/Source/JavaScriptCore/runtime/DateConversion.cpp        2018-02-27 16:47:25.182835822 +0100
+++ git.patched/Source/JavaScriptCore/runtime/DateConversion.cpp        2018-02-27 16:59:22.741504563 +0100
@@ -107,7 +107,8 @@
 #if OS(WINDOWS)
             TIME_ZONE_INFORMATION timeZoneInformation;
             GetTimeZoneInformation(&timeZoneInformation);
-            const WCHAR* timeZoneName = t.isDST() ? timeZoneInformation.DaylightName : timeZoneInformation.StandardName;
+            const WCHAR* winTimeZoneName = t.isDST() ? timeZoneInformation.DaylightName : timeZoneInformation.StandardName;
+            String timeZoneName(reinterpret_cast<const UChar*>(winTimeZoneName));
 #else
             struct tm gtm = t;
             char timeZoneName[70];
diff -ruN git/Source/WebKit2/Shared/API/c/WKString.cpp git.patched/Source/WebKit2/Shared/API/c/WKString.cpp
--- git/Source/WebKit2/Shared/API/c/WKString.cpp        2018-02-27 16:47:25.182835822 +0100
+++ git.patched/Source/WebKit2/Shared/API/c/WKString.cpp        2018-02-27 17:01:07.673903932 +0100
@@ -61,7 +61,7 @@
     unsigned unsignedBufferLength = std::min<size_t>(bufferLength, std::numeric_limits<unsigned>::max());
     auto substring = toImpl(stringRef)->stringView().substring(0, unsignedBufferLength);

-    substring.getCharactersWithUpconvert(static_cast<UChar*>(buffer));
+    substring.getCharactersWithUpconvert(reinterpret_cast<UChar*>(buffer));
     return substring.length();
 }
