https://github.com/WebPlatformForEmbedded/WPEBackend-rdk/pull/12

diff --git a/src/wayland-egl/CMakeLists.txt b/src/wayland-egl/CMakeLists.txt
index 28eba9e..aa56690 100644
--- a/src/wayland-egl/CMakeLists.txt
+++ b/src/wayland-egl/CMakeLists.txt
@@ -1,16 +1,19 @@
+find_package(EGL REQUIRED)
 find_package(Wayland REQUIRED)
 find_package(WaylandEGL REQUIRED)
 
-add_definitions(-DBACKEND_WAYLAND_EGL=1)
+add_definitions(-DBACKEND_WAYLAND_EGL=1 ${EGL_DEFINITIONS})
 
 list(APPEND WPE_PLATFORM_INCLUDE_DIRECTORIES
     "${CMAKE_SOURCE_DIR}/src/wayland"
     "${CMAKE_SOURCE_DIR}/src/wayland/protocols"
+    ${EGL_INCLUDE_DIRS}
     ${WAYLAND_INCLUDE_DIRS}
     ${WAYLAND_EGL_INCLUDE_DIRS}
 )
 
 list(APPEND WPE_PLATFORM_LIBRARIES
+    ${EGL_LIBRARIES}
     ${WAYLAND_LIBRARIES}
     ${WAYLAND_EGL_LIBRARIES}
 )
diff --git a/src/wayland-egl/view-backend.cpp b/src/wayland-egl/view-backend.cpp
index 6c44d76..33b0fca 100644
--- a/src/wayland-egl/view-backend.cpp
+++ b/src/wayland-egl/view-backend.cpp
@@ -108,7 +108,16 @@ void ViewBackend::handleMessage(char* data, size_t size)
 
 void ViewBackend::initialize()
 {
-    wpe_view_backend_dispatch_set_size( backend, WIDTH, HEIGHT );
+    uint32_t w = WIDTH, h = HEIGHT;
+    char *tmp;
+
+    if (tmp = std::getenv("WPE_INIT_VIEW_WIDTH"))
+        w = atoi(tmp);
+
+    if (tmp = std::getenv("WPE_INIT_VIEW_HEIGHT"))
+        h = atoi(tmp);
+
+    wpe_view_backend_dispatch_set_size( backend, w, h );
 }
 
 void ViewBackend::ackBufferCommit()
