From ea6475e893705fbf5c2144f9f83bde370ce1a9e2 Mon Sep 17 00:00:00 2001
From: Carlos Alberto Lopez Perez <clopez@igalia.com>
Date: Tue, 3 Jul 2018 16:16:47 +0200
Subject: [PATCH] Install a libWPEBackend-default.so symlink with fdo

---
 CMakeLists.txt | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3d93ded..92a102a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -168,6 +168,13 @@ target_include_directories(cogplatform-fdo PUBLIC wayland ${COGPLATFORM_FDO_INCL
 target_compile_options(cogplatform-fdo PUBLIC ${COGPLATFORM_FDO_CFLAGS})
 target_link_libraries(cogplatform-fdo cogcore ${COGPLATFORM_FDO_LDFLAGS})
 
+# Create a libWPEBackend-default.so symlink to libWPEBackend-fdo-0.1.so.0 and install it.
+add_custom_command(TARGET cogplatform-fdo
+                   POST_BUILD
+                   COMMAND ${CMAKE_COMMAND} -E create_symlink libWPEBackend-fdo-0.1.so.0 ${CMAKE_CURRENT_BINARY_DIR}/libWPEBackend-default.so
+)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libWPEBackend-default.so DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
 install(TARGETS cogplatform-fdo
     DESTINATION ${CMAKE_INSTALL_LIBDIR}
     COMPONENT "runtime"
-- 
2.11.0

