--- a/meson.build
+++ b/meson.build
@@ -3,11 +3,10 @@ project(
 	'c',
 	version: '0.12.0',
 	license: 'MIT',
-	meson_version: '>=0.54.0',
+	meson_version: '>=0.52.0',
 	default_options: [
 		'c_std=c11',
 		'warning_level=2',
-		'werror=true',
 	],
 )
 
@@ -172,7 +171,7 @@ wlroots = declare_dependency(
 	include_directories: wlr_inc,
 )
 
-meson.override_dependency('wlroots', wlroots)
+#meson.override_dependency('wlroots', wlroots)
 
 summary({
 	'systemd': conf_data.get('WLR_HAS_SYSTEMD', 0),
--- a/protocol/meson.build
+++ b/protocol/meson.build
@@ -1,6 +1,11 @@
 wl_protocol_dir = wayland_protos.get_variable(pkgconfig: 'pkgdatadir')
 
-wayland_scanner_dep = dependency('wayland-scanner', required: false, native: true)
+if meson.is_cross_build()
+	wayland_scanner_dep = disabler()
+else
+	wayland_scanner_dep = dependency('wayland-scanner', required: false, native: true)
+endif
+
 if wayland_scanner_dep.found()
 	wayland_scanner = find_program(
 		wayland_scanner_dep.get_variable(pkgconfig: 'wayland_scanner'),
